go/worker/registration/worker.go

```diff
@@ -130,108 +148,149 @@ func (r *Registration) doNodeRegistration() {
 					// and abort early.
 					return
 				}
-				r.logger.Error("failed to re-register node",
+				w.logger.Error("failed to re-register node",
 					"err", err,
 				)
 				continue
 			}
 			if first {
-				close(r.regCh)
+				close(w.regCh)
 				first = false
 			}
 		}
 	}
+
+	// Loop broken; shutdown requested.
+	publicKey := w.identity.NodeSigner.Public()
+
+	regCh, sub := w.registry.WatchNodes()
+	defer sub.Close()
+
+	// Check if the node is already deregistered.
+	node, err := w.registry.GetNode(w.ctx, publicKey, 0)
+	if err != nil {
+		w.logger.Error("can't get this node from the registry during shutdown wait",
```

Comment:
Hm, when a node expires, it will be removed from the registry, so this is normal (if error is `registry.ErrNoSuchNode`)?